src/OrleansServiceBus/Providers/Streams/EventHub/EventHubAdapterReceiver.cs

```diff
@@ -174,15 +178,15 @@ public Task Initialize(TimeSpan timeout)
             TimeSpan oldest = dequeueTimeUtc - messages[0].EnqueuedTimeUtc;
             TimeSpan newest = dequeueTimeUtc - messages[messages.Count - 1].EnqueuedTimeUtc;
 #endif
-            monitor.TrackAgeOfMessagesRead(oldest, newest);
+            monitor?.TrackMessagesReceived(messages.Count, oldest, newest);
 
             foreach (EventData message in messages)
             {
                 StreamPosition streamPosition = cache.Add(message, dequeueTimeUtc);
                 batches.Add(new StreamActivityNotificationBatch(streamPosition.StreamIdentity.Guid,
                     streamPosition.StreamIdentity.Namespace, streamPosition.SequenceToken));
             }
-
+            cache.CacheMonitor?.TrackMessageAdded(messages.Count);
```

Comment:
Suggest adding an Add(List<EventData> messages) to the IEventHubQueueCache, rather than exposing the monitor.